meshpy/mesh_creation_functions/beam_basic_geometry.py

```diff
@@ -91,7 +92,10 @@ def create_beam_mesh_line(
     t1 = direction / line_length
 
     # Check if the z or y axis are larger projected onto the direction.
-    if abs(np.dot(t1, [0, 0, 1])) < abs(np.dot(t1, [0, 1, 0])):
+    # The tolerance is used here to ensure that round-off changes in the last digits of
+    # the floating point values don't switch the case. This increases the robustness in
+    # testing.
+    if abs(np.dot(t1, [0, 0, 1])) < abs(np.dot(t1, [0, 1, 0])) - mpy.eps_quaternion:
```

Comment:
Some changes in the last digits of the vectors caused a switch in this `if` clause, thus the failing tests. Should now be working more robustly. 

@davidrudlstorfer I had to change the reference results, please check if the created mesh still looks as you expect it (especially the rotation of the triad in the cross section direction). Otherwise, we have to something different.

Reply:
The changes and resulting geometry looks good! Thanks for digging deeper and helping out here 🙏🏻